rs/sdk/src/api/command/command_handler.rs

```diff
@@ -8,24 +8,24 @@ use async_trait::async_trait;
 use std::sync::LazyLock;
 use std::{collections::HashMap, sync::Arc};
 
-pub struct CommandHandlerRegistry<R> {
-    commands: HashMap<String, Box<dyn CommandHandler<R>>>,
+pub struct CommandHandlerRegistry<R, S> {
+    commands: HashMap<String, Box<dyn CommandHandler<R, S>>>,
     on_set_api_key: Option<Box<dyn Fn(String) -> CommandResponse + Send + Sync + 'static>>,
     oc_client_factory: Arc<ClientFactory<R>>,
 }
 
 static SET_API_KEY_PARAMS: LazyLock<Vec<BotCommandParam>> = LazyLock::new(set_api_key_params);
 
-impl<R> CommandHandlerRegistry<R> {
-    pub fn new(oc_client_factory: Arc<ClientFactory<R>>) -> CommandHandlerRegistry<R> {
+impl<R, S> CommandHandlerRegistry<R, S> {
```

Comment:
If we were to hold the state on the `CommandHandlerRegistry` then we could set a default for S as `S = ()`, and then expose a `with_shared_state` which would hand the value to `CommandHandlerRegistry` and also set the S generic type on the returned instance.